### PR TITLE
revise lakemodel

### PR DESCRIPTION
**Pull request by arnavs**
Cleanup per https://github.com/QuantEcon/lecture-source-jl/issues/448.

### CODE REVIEW

File: rst_files/lake_model.rst [hunk 1]
```diff
@@ -217,60 +217,76 @@ Setup
...
         return x_path
     end
 
+Let's observe these matrices for the baseline model
+
 .. code-block:: julia
 
     lm = LakeModel()
-    lm.α
+    matrices = A_matrices(lm)
```
Comment (jlperla) [inline rst_files/lake_model.rst]: Having a named tuple called "matrices" is kind of weird and a step away from the math.

Why not just splat the results directly into variables and use them directly?

File: rst_files/lake_model.rst [hunk 2]
```diff
@@ -217,60 +217,76 @@ Setup
...
         for t in 1:T
             X_path[:, t] = X
-            X = lm.A * X
+            X = A * X
         end
         return X_path
     end
 
     function simulate_rate_path(lm, x0, T)
+        A_hat = A_matrices(lm).A_hat
```
Comment (jlperla) [inline rst_files/lake_model.rst]:
```
@unpack A_hat = A_matrices(lm)
```
for example....

File: rst_files/lake_model.rst [hunk 3]
```diff
@@ -217,60 +217,76 @@ Setup
 
 .. code-block:: julia
 
-    function LakeModel(;λ = 0.283, α = 0.013, b = 0.0124, d = 0.00822)
+    LakeModel = @with_kw (λ = 0.283, α = 0.013, b = 0.0124, d = 0.00822)
+
+    function A_matrices(lm)
```
Comment (jlperla) [inline rst_files/lake_model.rst]: I don't like the name.  

Maybe `transition_matrices` instead

File: rst_files/lake_model.rst [hunk 4]
```diff
@@ -217,60 +217,76 @@ Setup
...
+        @unpack λ, α, b, d = lm
         g = b - d
-        A = [(1 - λ) * (1 - d) + b (1 - d) * α + b   # carry out intermediate calculations
-            (1 - d) * λ            (1 - d) * (1 - α)]
+        A = [(1 - λ) * (1 - d) + b      (1 - d) * α + b
+            (1 - d) * λ                 (1 - d) * (1 - α)]
```
Comment (jlperla) [inline rst_files/lake_model.rst]: Need equation reference comment here and everywhere.

File: rst_files/lake_model.rst [hunk 5]
```diff
@@ -217,60 +217,76 @@ Setup
...
         A_hat = A ./ (1 + g)
```
Comment (jlperla) [inline rst_files/lake_model.rst]: Change `A_hat` to `Â` throughout.

This is a general comment.  For any annotations of ascii letters (e.g. bar, hat, etc.) use the appropriate glyph.  Sadly it doesn't work with annotations on greek letters/etc.

File: rst_files/lake_model.rst (hunk 1 above)
Comment (arnavs) [reply rst_files/lake_model.rst]: Done. I'll also add a note on this to the style guide (i.e., try not to create "fake" objects that don't correspond to something in the math or economics).

File: rst_files/lake_model.rst (hunk 2 above)
Comment (arnavs) [reply rst_files/lake_model.rst]: Done. I just felt weird using `@unpack` for one variable.

File: rst_files/lake_model.rst (hunk 3 above)
Comment (arnavs) [reply rst_files/lake_model.rst]: Done.

File: rst_files/lake_model.rst (hunk 5 above)
Comment (arnavs) [reply rst_files/lake_model.rst]: Done.

File: rst_files/lake_model.rst (hunk 4 above)
Comment (arnavs) [reply rst_files/lake_model.rst]: Sounds good. So unless you have other concerns, I'll merge this (might as well get this fix up) and create a separate issue for the equation numbering (?). If the thing above is too much work, we can always come back to this.

**Comment by arnavs** [comment]
Merging as it incorporates feedback. Will wait to hear from @mmcky about eqn #s.

**Pull request merged**